### ISSUE TITLE
Tool `extract dataset from a list` does not work

### ISSUE THREAD

**Issue by mmiladi** [galaxyproject/galaxy#11756]
Hello,

Following on the discussion here https://github.com/galaxyproject/galaxy/issues/11199, the Collection operation tool extract dataset from a list generates a popup html message window for the error below. I'm using the EU server and the problem occurs for any collection input. 

```
The server could not complete the request. Please contact the Galaxy Team if this error persists. Error executing tool with id '__EXTRACT_DATASET__': 'str' object has no attribute 'copy'

{
    "history_id": "fb9f7c937f67acb2",
    "tool_id": "__EXTRACT_DATASET__",
    "tool_version": "1.0.0",
    "inputs": {
        "input": {
            "values": [
                {
                    "id": "be88c7d0063e348d",
                    "hid": 409,
                    "name": "data 395, data 204, and others (filtered)",
                    "src": "hdca",
                    "tags": [
                        "name:subsampling",
                        "name:calu_fr",
                        "name:hg38",
                        "name:fan66494_pass_9329137c_pass.fastq"
                    ]
                }
            ],
            "batch": false
        },
        "which|which_dataset": "first"
    }
}
```

**Comment by mvdbeek** [comment]
Can you please share an example, because it works fine for me, but I'm sure there is some specificity to this.

**Comment by jmchilton** [comment]
Probably it was this https://github.com/galaxyproject/galaxy/pull/11741/files ? I think this happens if dataset elements have nametags. I think it will be fixed with an update to the latest 21.01.
